tests/php/Control/CookieTest.php

```diff
@@ -201,4 +206,57 @@ public function testForceExpiry()
         $this->assertEmpty(Cookie::get('newCookie'));
         $this->assertEmpty(Cookie::get('newCookie', false));
     }
+
+    /**
+     * Check whether warnings are correctly logged for insecure requests
+     */
+    public function testValidateSameSite(): void
+    {
+        // Throw an exception when a warning is logged so we can catch it
+        $mockLogger = $this->getMockBuilder(Logger::class)->setConstructorArgs(['testLogger'])->getMock();
+        $catchMessage = 'A warning was logged';
+        $mockLogger->expects($this->once())
+            ->method('warning')
+            ->willThrowException(new Exception($catchMessage));
+        Injector::inst()->registerService($mockLogger, LoggerInterface::class);
+
+        // Only samesite === 'None' should log a warning on insecure requests
+        Director::config()->set('alternate_base_url', 'http://insecure.example.com/');
```

Comment:
Split this one off to a new test method and use 
`$this->expectException(MyException::class);`
`$this->expectExceptionMessage($someMessage);`

Reply:
I'll do the same for the similar session test.